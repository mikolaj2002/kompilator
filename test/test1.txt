# Test 1. Arithmetic operations

PROGRAM IS
    a, b, result
IN
    # add
    a := 0;
    b := 0;
    result := a + b;
    WRITE result;
    a := 0;
    b := 4;
    result := a + b;
    WRITE result;
    a := 10;
    b := 0;
    result := a + b;
    WRITE result;
    a := 10;
    b := 12;
    result := a + b;
    WRITE result;

    # sub
    a := 0;
    b := 0;
    result := a - b;
    WRITE result;
    a := 0;
    b := 4;
    result := a - b;
    WRITE result;
    a := 10;
    b := 0;
    result := a - b;
    WRITE result;
    a := 10;
    b := 12;
    result := a - b;
    WRITE result;
    a := 15;
    b := 12;
    result := a - b;
    WRITE result;

    # mul
    a := 0;
    b := 0;
    result := a * b;
    WRITE result;
    a := 0;
    b := 4;
    result := a * b;
    WRITE result;
    a := 10;
    b := 0;
    result := a * b;
    WRITE result;
    a := 10;
    b := 12;
    result := a * b;
    WRITE result;
    a := 15;
    b := 12;
    result := a * b;
    WRITE result;

    # div
    a := 0;
    b := 0;
    result := a / b;
    WRITE result;
    a := 0;
    b := 4;
    result := a / b;
    WRITE result;
    a := 10;
    b := 0;
    result := a / b;
    WRITE result;
    a := 10;
    b := 1;
    result := a / b;
    WRITE result;
    a := 2;
    b := 12;
    result := a / b;
    WRITE result;
    a := 27;
    b := 3;
    result := a / b;
    WRITE result;
    a := 20;
    b := 6;
    result := a / b;
    WRITE result;

    # mod
    a := 0;
    b := 0;
    result := a % b;
    WRITE result;
    a := 0;
    b := 4;
    result := a % b;
    WRITE result;
    a := 10;
    b := 0;
    result := a % b;
    WRITE result;
    a := 10;
    b := 1;
    result := a % b;
    WRITE result;
    a := 2;
    b := 12;
    result := a % b;
    WRITE result;
    a := 27;
    b := 3;
    result := a % b;
    WRITE result;
    a := 20;
    b := 6;
    result := a % b;
    WRITE result;
END