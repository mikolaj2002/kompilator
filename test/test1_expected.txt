0
4
10
22
0
0
10
0
3
0
0
0
120
180
0
0
0
10
0
9
3
0
0
0
0
2
0
2
